httpbakery/discharge.go

```diff
@@ -100,6 +101,18 @@ func (d *dischargeHandler) serveDischarge1(p httprequest.Params) (interface{}, e
 	// TODO(rog) pass location into discharge
 	// location := p.Request.Form.Get("location")
 
+	if id[0] == 'A' {
+		// The id is base64 encoded, but contains a version byte
+		// of 0. Technically this byte could also be 1, 2, or 3;
+		// but these aren't defined yet. The canonical value for
+		// caveat ids of this type is unencoded so decode the
+		// base64 now.
+		id1, err := base64.URLEncoding.DecodeString(id)
+		if err != nil {
+			return nil, badRequestErrorf("cannot decode id: %s", err)
+		}
+		id = string(id1)
+	}
 	var resp dischargeResponse
 	m, err := d.svc.Discharge(bakery.ThirdPartyCheckerFunc(checker), id)
```

Comment:
That would be a backwards incompatible API change (which might be necessary anyway, but I'd like to avoid if possible)